src/texture_wrapper.rs

```diff
@@ -198,15 +272,42 @@ impl TexWrap {
 
                 let sub_img =
                     imageops::crop_imm(image, tex_start_x, tex_start_y, tex_width, tex_height);
-                let my_img = sub_img.to_image();
-                let tex = texture_generator_function(
-                    gfx,
-                    my_img.as_ref(),
-                    my_img.width(),
-                    my_img.height(),
-                    settings,
-                    allow_mipmap,
-                );
+
+                let tex: Option<Texture>;
+                match image.color() {
```

Comment:
I think you can match `image` directly here. This would prevent you from having to unwrap() inside the block

Reply:
Done!